### ISSUE TITLE
VkSurfaceCapabilitiesKHR::currentTransform is not required to be one of the supported transforms

### ISSUE THREAD

**Issue by zopsicle** [KhronosGroup/Vulkan-Docs#2440]
The spec should probably require of conforming implementations that currentTransform is one of the bits in supportedTransforms.

**Comment by cubanismo** [comment]
We investigated this internally and it was determined this requirement was purposely not included. While that seems counter-intuitive with only the context above, a reading of all the language related to these fields in the spec and what they represent makes things clearer. The reasoning is as follows:

* `currentTransform` reports the orientation of the viewing environment relative to the natural orientation of the presentation engine. E.g., for a user viewing a traditional desktop on a non-rotated display, this would be `VK_SURFACE_TRANSFORM_IDENTITY_BIT_KHR`. For a phone that has a natural orientation corresponding to "portrait" usage that has been rotated to "landscape" orientation, this would be `VK_SURFACE_TRANSFORM_ROTATE_90_BIT_KHR`.
* `supportedTransforms` reports to the pre-transformed orientations the presentation engine can accept (Via `VkSwapchainCreateInfoKHR::preTransform`) for the surface in its current state. For a viewing environment in a 90-degree rotated state, this might be expected to include only `VK_SURFACE_TRANSFORM_ROTATE_90_BIT_KHR`, meaning the presentation engine need not alter the image as part of the presentation operation, but in practice it would often be limited to only `VK_SURFACE_TRANSFORM_IDENTITY_BIT_KHR`, meaning the presentation engine will need to perform a 90 degree rotation as part of the presentation operation. The latter case would generally occur when presenting to a target, such as a window in the X11 windowing system, that has no way for a client to express an already-rotated surface, and hence always performs such a rotation when accepting presentation requests while running in a rotated state.

In other words, `supportedTransforms` both indirectly specifies the rotation operations a presentation engine can perform, when compared to `currentTransform`, and also specifies the orientation Vulkan can express to the presentation engine for a given surface, while `currentTransform` merely reports the state of the surface. Applications are required to interpret these fields and determine whether they can or must perform any pre-present rotation of their own for a given surface+swapchain.
